src/util/metrics/mod.rs

```diff
@@ -60,14 +60,15 @@ pub fn run_prometheus(
     Some(handler)
 }
 
-pub fn dump() -> String {
+pub fn dump() -> Result<String, Error> {
     let mut buffer = vec![];
     let encoder = TextEncoder::new();
     let metric_familys = prometheus::gather();
     for mf in metric_familys {
         if let Err(e) = encoder.encode(&[mf], &mut buffer) {
-            warn!("ignore prometheus encoding error: {:?}", e);
+            warn!("prometheus encoding error: {:?}", e);
+            return Err(e);
         }
     }
-    String::from_utf8(buffer).unwrap()
+    Ok(String::from_utf8(buffer).unwrap())
```

Comment:
Maybe we should avoid `unwrap()`.

Reply:
Any suggestions?

Reply:
How about just `Ok(String::from_utf8(buffer)?)`?